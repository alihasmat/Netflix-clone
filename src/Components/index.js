export { default as Accordian } from "./Accordian/Accordian";
export { default as Card } from "./Card/Card";
export { default as Feature } from "./Feature/Feature";
export { default as Footer } from "./Footer/Footer";
export { default as Form } from "./Form/Form";
export { default as Header } from "./Header/Header";
export { default as Jumbotron } from "./Jumbotron/Jumbotron";
export { default as Loading } from "./Loading/Loading";
export { default as Player } from "./Player/Player";
export { default as Profiles } from "./Profiles/Profiles";
export { default as Search } from "./Search/Search";
