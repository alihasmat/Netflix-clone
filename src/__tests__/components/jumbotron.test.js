import { getByAltText, render } from "@testing-library/react";
import { Jumbotron } from "../../Components";
import JumboData from "../../fixtures/jumbo.json";

describe("<Jumbotron />", () => {
  it(`renders <Jumbotron /> with populated data`, () => {
    const { container, getByText, getByTestId, getAllByAltText } = render(
      <Jumbotron.Container>
        {JumboData.map((item) => (
          <Jumbotron key={item.id} direction={item.direction}>
            <Jumbotron.Pane>
              <Jumbotron.Title>{item.title}</Jumbotron.Title>
              <Jumbotron.SubTitle>{item.subTitle}</Jumbotron.SubTitle>
            </Jumbotron.Pane>
            <Jumbotron.Pane>
              <img
                src={item.image}
                alt={item.alt}
                data-testid={`${item.id}-jumbo-image`}
              />
            </Jumbotron.Pane>
          </Jumbotron>
        ))}
      </Jumbotron.Container>
    );
    expect(getByText("Enjoy on your TV.")).toBeTruthy();
    expect(getAllByAltText("Tiger King on Netflix")).toBeTruthy();
    expect(
      getByText(
        "Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more."
      )
    ).toBeTruthy();
    expect(getByTestId("1-jumbo-image")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
