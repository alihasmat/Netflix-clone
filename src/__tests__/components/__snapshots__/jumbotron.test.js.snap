// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders <Jumbotron /> with populated data 1`] = `
<div
  class="style__Container-sc-1podvso-2"
>
  <div
    class="style__Item-sc-1podvso-0 kJcZnJ"
  >
    <div
      class="style__Inner-sc-1podvso-1 dzEnHV"
      direction="row"
    >
      <div
        class="style__Pane-sc-1podvso-3 bnTcCt"
      >
        <h1
          class="style__Title-sc-1podvso-4 fOyRNM"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="style__SubTitle-sc-1podvso-5 gzYvZu"
        >
          Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.
        </h2>
      </div>
      <div
        class="style__Pane-sc-1podvso-3 bnTcCt"
      >
        <img
          alt="Tiger King on Netflix"
          data-testid="1-jumbo-image"
          src="images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="style__Item-sc-1podvso-0 kJcZnJ"
  >
    <div
      class="style__Inner-sc-1podvso-1 hPpEjk"
      direction="row-reverse"
    >
      <div
        class="style__Pane-sc-1podvso-3 bnTcCt"
      >
        <h1
          class="style__Title-sc-1podvso-4 fOyRNM"
        >
          Download your programmes to watch on the go.
        </h1>
        <h2
          class="style__SubTitle-sc-1podvso-5 gzYvZu"
        >
          Save your data and watch all your favourites offline.
        </h2>
      </div>
      <div
        class="style__Pane-sc-1podvso-3 bnTcCt"
      >
        <img
          alt="Tiger King on Netflix"
          data-testid="2-jumbo-image"
          src="images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="style__Item-sc-1podvso-0 kJcZnJ"
  >
    <div
      class="style__Inner-sc-1podvso-1 dzEnHV"
      direction="row"
    >
      <div
        class="style__Pane-sc-1podvso-3 bnTcCt"
      >
        <h1
          class="style__Title-sc-1podvso-4 fOyRNM"
        >
          Watch everywhere.
        </h1>
        <h2
          class="style__SubTitle-sc-1podvso-5 gzYvZu"
        >
          Stream unlimited films and TV programmes on your phone, tablet, laptop and TV without paying more.
        </h2>
      </div>
      <div
        class="style__Pane-sc-1podvso-3 bnTcCt"
      >
        <img
          alt="Money Heist on Netflix"
          data-testid="3-jumbo-image"
          src="images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
